nodejs/awsx/apigateway/api.ts

```diff
@@ -581,6 +622,97 @@ function addEventHandlerRouteToSwaggerSpec(
     }
 }
 
+function addAuthorizersToSwagger(swagger: SwaggerSpec, authorizers: authorizer.SecurityDefinition[]): string[] {
+    const authNames: string[] = [];
+    for (const auth of authorizers) {
+        swagger["securityDefinitions"] = swagger["securityDefinitions"] || {};
+
+        const suffix = Object.keys(swagger["securityDefinitions"]).length;
+        const authName = auth.authorizerName || `authorizer-${suffix}`;
+
+        const securityDef: SecurityDefinition = {
+            type: "apiKey",
+            name: auth.parameterName,
+            in: auth.parameterLocation,
+            "x-amazon-apigateway-authtype": auth.authType,
+            "x-amazon-apigateway-authorizer": getLambdaAuthorizer(authName, auth.authorizer),
+        };
+
+        // Add security definition if it's a new authorizer
+        if (!swagger["securityDefinitions"][authName]) {
+            authNames.push(authName);
+            swagger["securityDefinitions"][authName] = securityDef;
+        } else if (swagger["securityDefinitions"][authName] !== securityDef) {
+
+            // Error if security definition with the name already exists and its different from the existing definition
+            throw new Error("Different authorizers with same name " + authName);
+        }
+    }
+    return authNames;
+}
+
+function getLambdaAuthorizer(authorizerName: string, lambdaAuthorizer: authorizer.LambdaAuthorizer): LambdaAuthorizer {
+    if (authorizer.isLambdaInfo(lambdaAuthorizer.authorizer)) {
+        return {
+            type: lambdaAuthorizer.type,
+            authorizerUri: lambdaAuthorizer.authorizer.authorizerUri,
+            authorizerCredentials: lambdaAuthorizer.authorizer.authorizerCredentials,
+            identitySource: lambdaAuthorizer.identitySource,
+            identityValidationExpression: lambdaAuthorizer.identityValidationExpression,
+            authorizerResultTtlInSeconds: lambdaAuthorizer.authorizerResultTtlInSeconds,
+        };
```

Comment:
looks like this could be entirely replaced with `return { type: lambdaAuthorizer.type, ...lambdaAuthorizer.authorizer }`